3_javascript/javascript1/week2/script.js

```diff
@@ -0,0 +1,129 @@
+/** Smart-ease - Goes Global! **/
+
+function getFullname(Gender, firstName, surName, useFormalName) {
+  let fullName = "";
+
```

Comment:
You can simplify the code by checking for `useFormalName` first and then determining the title based on gender. This way, you can eliminate one condition block because the non-formal name is the same for both men and women.

```javascript
function getFullname(Gender, firstName, surName, useFormalName) {
  let fullName = "";

  // Check if formal name is requested
  if (useFormalName) {
    if (Gender === "female") {
      // Use "Lady" for a formal female title
      fullName = `Lady ${firstName} ${surName}`;
    } else {
      // Use "Lord" for a formal male title
      fullName = `Lord ${firstName} ${surName}`;
    }
  } else {
    // Use only the first and last name for informal names
    fullName = `${firstName} ${surName}`;
  }

  return fullName;
}
```

This code structure reduces redundancy and makes the logic more straightforward. If `useFormalName` is `true`, it will add the appropriate title based on gender, and if `useFormalName` is `false`, it will only use the first and last names for an informal name.